mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -170,15 +294,59 @@ mod tests {
     #[serial(env_settings)]
     fn from_file_reads_nested_bridge_settings() {
         let settings = Settings::from_file("tests/config.json").unwrap();
+        let upstream = settings.upstream().unwrap();
+
+        assert_eq!(upstream.name(), "upstream");
+        assert_eq!(upstream.address(), "edge1");
+
+        match upstream.credentials() {
+            Credentials::Provider(provider) => {
+                assert_eq!(provider.device_id().unwrap(), "d1");
+                assert_eq!(provider.module_id().unwrap(), "mymodule");
+                assert_eq!(provider.generation_id().unwrap(), "321");
+                assert_eq!(provider.workload_uri().unwrap(), "uri");
+            }
+            _ => panic!("Expected provider settings"),
+        };
+    }
+
+    #[test]
+    #[serial(env_settings)]
+    fn from_file_reads_remotes_settings() {
+        let settings = Settings::from_file("tests/config.json").unwrap();
+        let len = settings.remotes().len();
+
+        assert_eq!(len, 1);
+        let remote = settings.remotes().first().unwrap();
+        assert_eq!(remote.name(), "r1");
+        assert_eq!(remote.address(), "remote");
+        assert_eq!(remote.keep_alive().as_secs(), 60);
+        assert_eq!(remote.clean_session(), false);
+
+        match remote.credentials() {
+            Credentials::PlainText(auth_settings) => {
+                assert_eq!(auth_settings.username(), "mymodule");
+                assert_eq!(auth_settings.password(), "pass");
+                assert_eq!(auth_settings.client_id(), "client");
+            }
+            _ => panic!("Expected plaintext settings"),
+        };
+    }
+
+    #[test]
+    #[serial(env_settings)]
+    fn from_file_reads_messages_settings() {
+        let settings = Settings::from_file("tests/config.json").unwrap();
 
-        assert_eq!(
-            settings.nested_bridge().gateway_hostname().unwrap(),
-            "edge1"
-        );
-        assert_eq!(settings.nested_bridge().device_id().unwrap(), "d1");
-        assert_eq!(settings.nested_bridge().module_id().unwrap(), "mymodule");
-        assert_eq!(settings.nested_bridge().generation_id().unwrap(), "321");
-        assert_eq!(settings.nested_bridge().workload_uri().unwrap(), "uri");
+        assert_eq!(settings.messages.max_count(), 10);
+    }
+
+    #[test]
+    #[serial(env_settings)]
+    fn from_default_sets_messages_settings() {
+        let settings = Settings::new().unwrap();
+
+        assert_eq!(settings.messages.max_count(), 100);
     }
```

Comment:
I noticed that we have tests in the broker verifying the defaults. I don't think this is super useful, but others might disagree and recommend adding here.

Reply:
I think they are useful to make sure that defaults file is loaded

Reply:
Yes, it is a nice idea to check defaults, because of several sources of settings: env, default config file, impl Default, etc